packages/contracts/contracts/testing/ERC20Reward.sol

```diff
@@ -21,4 +22,46 @@ contract ERC20Reward is ERC20, ERC20Burnable, ERC20Permit, ERC7529 {
     function mint(address to, uint256 amount) public onlyRole(MINTER_ROLE) {
         _mint(to, amount);
     }
+
+    function redeem(address account, uint256 value, bytes memory signature) public {
+
+        // Regenerate the message 
+        bytes32 hash = ECDSA.toEthSignedMessageHash(
+            keccak256(abi.encodePacked(address(this), account, value))
+        );
+
+        // Get the signer of the message
+        address signer = _getSigner(hash, signature);
+
+        // If the caller has is an admin, check the signer against the target account to burn tokens from
+        // If the caller is not an admin, check that the function caller is the signer
+        if (hasRole(DEFAULT_ADMIN_ROLE, _msgSender())) {
+            require(signer == account, "ERC20Reward: Signer does not match account to redeem from");
+        } else {
+            require(signer == _msgSender(), "ERC20Reward: Caller did not sign the message"); 
```

Comment:
What a good catch! This needs an extra signer == _msgSender() && account == signer